translate.js

```diff
@@ -30,88 +30,31 @@ const helper = (xs) => ([x, position, y], val) =>
 const cornersHelper = helper(corners);
 const edgesHelper = helper(edges);
 
-const colorHelper = (colors, color, shade) => {
+const colorHelper = (colors, color, shade, opacityHelper) => {
   const base = colors[color];
-  return (
+  const colorValue = (
     base &&
     (typeof base === 'string' ? base : shade ? base[shade] : base['default'])
   );
+  return opacityHelper ? hexToRgb(colorValue, opacityHelper) : colorValue;
```

Comment:
This is looking way better.. although when I suggested this bringing the opacity stuff into this function I totally overlooked that there was both `--ow-bg--opacity` and `--ow-border-opacity` 🤦  I guess `--ow-text-opacity`, `--ow-divide-opacity`, `--ow-placeholder-opacity` and `--ow-ring-opacity` are going to work in the same way too!

So you were right in bringing in a variable for that. Lets stick with a similar approach as you have gone for here but what do you think to returning this:

```js
return `rgba(${hexToRgb(colorValue)}, var(${opacity}, 1))`
```

Then when we return colors values from translations, they come with a var and default value:

```js
out = {
  '--ow-bg-opacity': 1,
  'background-color': colorHelper(theme.colors, i[1], i[2], '--ow-bg-opacity')
}
```

I _think_ that this is how tailwind does it.. at least that is what I can deduct from looking at the styles on their docs from the inspector.

Reply:
Hmm, yeah. I did this with the idea of avoiding unnecessary color processing, but if every color is going to have the option of opacity this way (something I completely forgot about), a lot of this can be rewritten.

I haven't delved too deeply into the logic, but I'm assuming Oceanwind knows to apply something like `bg-opacity-x` after the the opacity var that'll come with each color with this change?

I'll get this written up.

Reply:
I think I've covered this fully now. Tests all pass, and I believe I've added tests for all the cases.

Reply:
> I'm assuming Oceanwind knows to apply something like bg-opacity-x after the the opacity var that'll come with each color

So I was making this same assumption 😅 but aftyer thinking about it a bit more.. we actually do a deep merge of properties (including vars) before handing the CSS-inJS object to otion. So it does require users of oceanwind to format rule sets like this:

```
bg-red-500 bg-opacity-50
```

Not like this:

```
bg-opacity-50 bg-red-500
```

This seems sensible from a user perspective and I'm hoping it won't cause too many issues.. I guess we can always re-write this slightly if it does.

Thanks for all your amendments here, this is looking really good now!